Assets/Scripts/Switch.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Switch : MonoBehaviour
+{
+    [SerializeField]
+    [Tooltip("The color of the switch. 0 if yellow, 1 if blue, 2 if red, 3 if green.")]
+    [Range(0, 3)]
+    private int switchColor;
+
+    [SerializeField]
+    [Tooltip("The puzzle this switch is connected to.\nMust have a script which implements the IPuzzle interface.")]
+    private GameObject connectedPuzzle;
```

Comment:
Why don't you make this of the type IPuzzle as opposed to GameObject? Enforce these rules, don't just mention them in text.

Reply:
I tried using IPuzzle at first, but from what I found, interfaces aren't serializable. I even tried adding the [Serializable] attribute before the class, but it didn't work. If there's a different way to do it that I didn't think of, I'd love to hear it, the current solution isn't preferable.